ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/PushPhysicalPlan.java

```diff
@@ -0,0 +1,152 @@
+ */
+public class PushPhysicalPlan {
+  // Subscribers always demand one row, so it shouldn't be that any buffering is required in
+  // practice.
+  private static final int CAPACITY = Integer.MAX_VALUE;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PullPhysicalPlan.class);
+
+  private final AbstractPhysicalOperator root;
+
+  private final LogicalSchema schema;
+  private final QueryId queryId;
+  private final ScalablePushRegistry scalablePushRegistry;
+  private final PushDataSourceOperator dataSourceOperator;
+  private final Context context;
+  private volatile boolean closed = false;
+  private long timer = -1;
+
+  public PushPhysicalPlan(
+      final AbstractPhysicalOperator root,
+      final LogicalSchema schema,
+      final QueryId queryId,
+      final ScalablePushRegistry scalablePushRegistry,
+      final PushDataSourceOperator dataSourceOperator,
+      final Context context
+  ) {
+    this.root = Objects.requireNonNull(root, "root");
+    this.schema = Objects.requireNonNull(schema, "schema");
+    this.queryId = Objects.requireNonNull(queryId, "queryId");
+    this.scalablePushRegistry =
+        Objects.requireNonNull(scalablePushRegistry, "scalablePushRegistry");
+    this.dataSourceOperator = dataSourceOperator;
+    this.context = context;
+  }
+
+  public BufferedPublisher<List<?>> execute() {
+    final Publisher publisher = new Publisher(context);
+    context.runOnContext(v -> open(publisher));
+    return publisher;
+  }
+
+  public boolean isClosed() {
+    return closed;
+  }
+
+  private void maybeNext(final Publisher publisher) {
+    List<?> row;
+    while ((row = (List<?>)next()) != null) {
+      if (dataSourceOperator.droppedRows()) {
+        closeInternal();
+        publisher.reportDroppedRows();
+        break;
+      } else {
+        publisher.accept(row);
+      }
+    }
+    if (!closed) {
+      if (timer >= 0) {
+        context.owner().cancelTimer(timer);
+      }
+      // Schedule another batch async
+      timer = context.owner()
+          .setTimer(100, timerId -> context.runOnContext(v -> maybeNext(publisher)));
+    } else {
+      publisher.close();
+    }
+  }
+
+  private void open(final Publisher publisher) {
+    VertxUtils.checkContext(context);
+    dataSourceOperator.setNewRowCallback(() -> context.runOnContext(v -> maybeNext(publisher)));
+    root.open();
+    maybeNext(publisher);
+  }
+
+  private Object next() {
+    VertxUtils.checkContext(context);
+    return root.next();
+  }
+
+  public void close() {
+    context.runOnContext(v -> closeInternal());
+  }
+
+  private void closeInternal() {
+    VertxUtils.checkContext(context);
+    closed = true;
+    root.close();
+  }
+
+  public AbstractPhysicalOperator getRoot() {
+    return root;
+  }
+
+  public QueryId getQueryId() {
+    return queryId;
+  }
+
+  public LogicalSchema getOutputSchema() {
+    return schema;
+  }
+
+  public ScalablePushRegistry getScalablePushRegistry() {
```

Comment:
This getter, and hence the member field, seems not used anywhere in this or the other PR?